docs/book/customer_and_shopuser.rst

```diff
@@ -0,0 +1,83 @@
+.. index::
+   single: Customer and ShopUser
+
+Customer and ShopUser
+=====================
+
+For handling customers of your system **Sylius** is using a combination of two entities - **Customer** and **ShopUser**.
+The difference between these two entities is simple:
+the **Customer** is unregistered and the **ShopUser** is registered in the system - they have an account.
+
+Customer
+--------
+
+The Customer entity was created to collect data about non-registered guests of the system- one that have been buying without having an account
+or that have somehow provided us their e-mail.
+
+How to create a Customer programmatically?
+''''''''''''''''''''''''''''''''''''''''''
+
+As usually, use a factory. The only required field for the Customer entity is ``email``, provide it before adding it to the repository.
+
+.. code-block:: php
+
+   /** @var CustomerInterface $customer */
+   $customer = $this->container->get('sylius.factory.customer')->createNew();
+
+   $customer->setEmail('customer@test.com');
+
+   $this->container->get('sylius.repository.customer')->add($customer);
+
+The Customer entity can of course hold other information besides an email, it can be for instance
+``billingAddress`` and ``shippingAddress``, ``firstName``, ``lastName`` or ``birthday``.
+
+.. note::
+
+   The relation between the Customer and User is bidirectional. Both entities hold a reference to each other.
+
+ShopUser
+--------
+
+ShopUser entity is  designed for customers that have registered in the system - they have an account with both e-mail and password.
+They can visit and modify their account.
+
+While creating new account the existence of the provided email in the system is checked - if the email was present - it will already have a Customer
+therefore the existing one will be assigned to the newly created ShopUser, if not - a new Customer will be created together with the ShopUser.
+
+How to create a ShopUser programmatically?
+''''''''''''''''''''''''''''''''''''''''''
+
+Assuming that you have a Customer (either retrieved from the reposiotry or a new one created) - use a factory to create
+a new ShopUser, assign the existing Customer and a password via the ``setPlainPassword()`` method.
+
+.. code-block:: php
+
+   /** @var ShopUserInterface $user */
+   $user = $this->container->get('sylius.factory.shop_user')->createNew();
+
+   // Now let's find a Customer by their e-mail:
+   /** @var CustomerInterface $customer */
+   $customer = $this->container->get('sylius.repository.customer`)->findOneBy(['email' => 'customer@test.com']);
+
+   // and assign it to the ShopUser
+   $user->setCustomer($customer);
+   $user->setPlainPassword('pswd');
+
+   $this->container->get('sylius.repository.shop_user')->add($user);
+
+Changing the ShopUser password
+''''''''''''''''''''''''''''''
+
+The already set password of a **ShopUser** can be easily changed via the ``setPlainPassword()`` method.
+
+.. code-block:: php
+
+   $user->getPlainPassword(); // returns 'pswd'
```

Comment:
It is not 100% true. AFAIR this method is valid only before persisting user to the database(or to the moment when a `eraseCredentials` function would be called). Usually, you are not able to get users plain password for security reasons. You are only allowed to get password via `getPassword` method, but this one will be encrypted.